lib/adapters/FileSystem.js

```diff
@@ -37,69 +38,63 @@ class FileSystem extends AbstractAdapter {
 	 * @param {module:@ui5/fs.tracing.Trace} trace Trace instance
 	 * @returns {Promise<module:@ui5/fs.Resource[]>} Promise resolving to list of resources
 	 */
-	_runGlob(patterns, options = {nodir: true}, trace) {
-		return new Promise((resolve, reject) => {
-			const opt = {
-				cwd: this._fsBasePath,
-				dot: true,
-				nodir: options.nodir
-			};
-
-			trace.globCall();
-			glob(patterns, opt).then((matches) => {
-				const promises = [];
-				if (!opt.nodir && patterns[0] === "") { // Match physical root directory
-					promises.push(new Promise((resolve, reject) => {
-						fs.stat(this._fsBasePath, (err, stat) => {
-							if (err) {
-								reject(err);
-							} else {
-								resolve(new Resource({
-									project: this._project,
-									statInfo: stat,
-									path: this._virBaseDir,
-									createStream: () => {
-										return fs.createReadStream(this._fsBasePath);
-									}
-								}));
-							}
-						});
-					}));
-				}
-
-				for (let i = matches.length - 1; i >= 0; i--) {
-					promises.push(new Promise((resolve, reject) => {
-						const fsPath = path.join(this._fsBasePath, matches[i]);
-						const virPath = (this._virBasePath + matches[i]);
-
-						// Workaround for not getting the stat from the glob
-						fs.stat(fsPath, (err, stat) => {
-							if (err) {
-								reject(err);
-							} else {
-								resolve(new Resource({
-									project: this._project,
-									statInfo: stat,
-									path: virPath,
-									createStream: () => {
-										return fs.createReadStream(fsPath);
-									}
-								}));
+	async _runGlob(patterns, options = {nodir: true}, trace) {
+		const opt = {
+			cwd: this._fsBasePath,
+			dot: true,
+			onlyFiles: options.nodir,
+			followSymlinkedDirectories: false
+		};
+		trace.globCall();
+
+		const promises = [];
+		if (!opt.onlyFiles && patterns[0] === "") { // Match physical root directory
+			promises.push(new Promise((resolve, reject) => {
+				fs.stat(this._fsBasePath, (err, stat) => {
+					if (err) {
+						reject(err);
+					} else {
+						resolve(new Resource({
+							project: this._project,
+							statInfo: stat,
+							path: this._virBaseDir,
+							createStream: () => {
+								return fs.createReadStream(this._fsBasePath);
 							}
-						});
-					}));
-				}
-
-				Promise.all(promises).then(function(results) {
-					const flat = Array.prototype.concat.apply([], results);
-					resolve(flat);
-				}, function(err) {
-					reject(err);
+						}));
+					}
 				});
-			}).catch((err) => {
-				log.error(err);
-			});
-		});
+			}));
+		}
+		if (patterns.length > 1 || patterns[0] !== "") {
```

Comment:
It took me a while to understand why this checks for `length > 1`, not `length > 0`.
This is related to the "physical root directory" matching above, right?

Reply:
Clarified:
```
		// Only glob if the only pattern is an empty string.
		// Since globby v8 or v9 empty glob patterns "" act like "**"
```